tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -434,6 +434,13 @@ decompress_chunk_create_tuple(DecompressChunkState *state)
 
 		ExecClearTuple(slot);
 
+		/* some technical columns may left unpopulated in the slot;
+		 * setting isnull prevents issues for row expressions */
+		for (i = 0; i < slot->tts_tupleDescriptor->natts; i++)
+		{
+			slot->tts_isnull[i] = true;
+		}
+
```

Comment:
Which columns exactly? Shouldn't `drop column battery_status` drop them as well?

Probably makes sense to add a test for whole-row var + compressed chunks with diffent columns, like in `compression_alter.sql`. I'm not sure how row type conversion works in this case, and why it even tries to read these columns that are not in the resulting row...

Reply:
for columns with types (`state->columns[i].type`) like `COUNT_COLUMN` and `SEQUENCE_NUM_COLUMN`  there is a place in the result desc; however its never populated; however when the row-tuple is read it tries to flatten it...

there is a testcase in the changeset which is able to trigger the issue without these changes

by the way: it returns the a much wider row than whats required; however it uses a by-name-lookup so its "not-a-problem" :D

Reply:
after the drop there is a 3rd column in `tts_tupleDescriptor` with the name : ` {data = "........pg.dropped.3........", '\000' <repeats 35 times>}` which is not populated

I've looked into fixing it differently but the removed column can be anywhere in the row - so its not that easy to erase it; we also have have a projection embedded around here which further complicate things

Reply:
The problem is the projections from the 2 chunks don't match. The compressed chunk output should be transformed to match that of the uncompressed chunk. 
Compare the plan below with select * from readings. We already have code that adjusts for select *. Need to check why that is not working for this case.
```
test=# explain verbose select c from readings c;
                                                                     QUERY PLAN                                                                     
----------------------------------------------------------------------------------------------------------------------------------------------------
 Append  (cost=1.01..43.76 rows=2850 width=32)
   ->  Custom Scan (DecompressChunk) on _timescaledb_internal._hyper_9_57_chunk c_1  (cost=1.01..1.01 rows=1000 width=32)
         Output: c_1.*
         ->  Seq Scan on _timescaledb_internal.compress_hyper_10_58_chunk  (cost=0.00..1.01 rows=1 width=44)
               Output: compress_hyper_10_58_chunk._ts_meta_count, compress_hyper_10_58_chunk."time", compress_hyper_10_58_chunk.battery_temperature
   ->  Seq Scan on _timescaledb_internal._hyper_9_59_chunk c_2  (cost=0.00..28.50 rows=1850 width=32)
         Output: c_2.*
(7 rows)
```

Reply:
See decompress_chunk.c for whole_row var handling

Reply:
I've debugged this a bit more and I'm now more convinced that this is the right solution; let me try to walk you thru why:
* decompression with uncompressed rows works by utilizing an Append node to concatenate the rows of 2 operators
* `Append` also has the requirement that the columns should match-up between its operands
  * in case there are exact columns requests this is not that big of an issue - they are being placed into the targetlist; and as long as the number/types of the targets are in sync I guess Append is happy to do its job
    * `note that` in this case we are selecting specific columns from the standard hypertable as well
  * if we have to select the whole row - it will do it by adding a Var with `attno=0`; there will be 1 column with a complex type...
* whole-rows for compression is handled by interpreting the `attno=0` and expanding them for `compressed_rel` ;  to help build the underlying SeqScan on the compressed table.
  * based on the actual scanned column list on the compressed table the `compression_map` is computed which will be there during execution to route inputs to outputs after decompression
* to make `Append` work the code actually piggybacks on the original rel ; by copying its targetlist for the decompression path
  * however this means that in the above path it will copy the row reference as well: `c_2.*`
  * this will make Append happy because there is no difference between the 2 operands output :+1: 
* the hypertable already have a column dropped!
  * there will be a `pg.dropped` column in the output rowtype of of `Seq Scan c_2` for `c_2.*`
  * because this rowtype was copied to be the output of the `DecompressChunk`; the deleted column will appear there as well...but since it was never really planned to get that column (whole_row stuff in decompress_chunk) - `compression_map` will not have any instructions how to fill it based on the actual input
* I think the removal of `pg.dropped.` columns is done internally by postgres automatically

I think append will not work correctly unless all branches return the same rowtype. And that's why I think the options are:
* go with the current fix
  * or it might be possible to pre-determine which columns will not have their `is_null` populated
* map: probably it might be possible to add some marker to the  decompress_map to handle unknown output columns and set is_null based on that
* wild1: if the input contains whole-row stuff; reference all columns directly for both the hypertable and for the compressed; move all expression evaluation into a new projection on top of the append
* wild2: push-in a projection for the standard seq-scan part to remove these removed columns
* other ideas?

Reply:
@gayyappan @akuzm

Reply:
@svenklemm  ; replying to your comment on the other PR [here](https://github.com/timescale/timescaledb/pull/5488#discussion_r1148865909) instead.

row schema for the compressed path is copied over from the non-compressed hypertable; which may have dropped columns.

* `CompressionInfo.chunk_rel` is the base hypertable relation
* decompression uses that to declare that it could a relation like that:
https://github.com/timescale/timescaledb/blob/22841abdf0d2c7b853386d8594d2a317df5f1e07/tsl/src/nodes/decompress_chunk/decompress_chunk.c#L1199
  * actually this kinda makes sense ; because for a compressed chunk that's the "closest thing" to what the compression will be producing

We could try other approaches...but setting the `isnull` bools to true by default can fix this segfault harmlessly without extensive changes